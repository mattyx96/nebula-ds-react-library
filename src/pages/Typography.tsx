const displayText = 'Lorem ipsum dolor';
const headerText = 'Lorem ipsum dolor sit amet consectetur';
const text = `Lorem ipsum dolor sit amet consectetur adipiscing elit sociis himenaeos, suscipit hac cum est aptent vitae etiam egestas mi, imperdiet nunc nisi netus lectus parturient convallis orci. Dignissim dis id massa commodo odio justo ad habitasse molestie primis, cubilia et vehicula malesuada convallis sodales nec cras purus eget nisi, venenatis velit dictum orci condimentum quis tincidunt litora diam.`;

export const TypographyPage = () => {
  return (
    <div className="flex flex-col gap-20 items-start max-w-full pb-20">
      <h1>h1: {displayText}</h1>
      <h2>h2: {displayText}</h2>
      <h3>h3: {headerText}</h3>
      <h4>h4: {headerText}</h4>
      <h5>h5: {headerText}</h5>
      <h6>h6: {headerText}</h6>
      <p>p: {text}</p>
      <span>span: {text}</span>
      <p><small>p - small: {text}</small></p>
      <small>small: {text}</small>
    </div>
  );
}
