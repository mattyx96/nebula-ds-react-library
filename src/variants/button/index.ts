import { VariantProps } from 'class-variance-authority';
import { buttonVariants } from './buttonVariants.ts';
import { iconButtonVariantsExtendsButton } from './iconButtonVariants.ts';

export type ButtonVariants = VariantProps<typeof buttonVariants>;
export type IconButtonVariants = VariantProps<
  typeof iconButtonVariantsExtendsButton
>;

export { buttonVariants, iconButtonVariantsExtendsButton };

export const buttonVariantVariants = [
  'filled',
  'outlined',
  'standard',
  'text',
] as const satisfies Omit<
  VariantProps<typeof buttonVariants>['variant'],
  'null' | 'undefined'
>;

export const buttonSizeVariants = ['L', 'M', 'S'] as const satisfies Omit<
  VariantProps<typeof buttonVariants>['size'],
  'null' | 'undefined'
>;

export const buttonRoundedVariants = [
  'Default',
  'R',
  'RTop',
  'RBottom',
  'L',
  'LTop',
  'LBottom',
] as const satisfies Omit<
  VariantProps<typeof buttonVariants>['rounded'],
  'null' | 'undefined'
>;
