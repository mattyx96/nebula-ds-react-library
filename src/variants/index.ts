export * from './button/buttonVariants.ts';
