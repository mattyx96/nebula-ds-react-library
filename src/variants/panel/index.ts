import {VariantProps} from "class-variance-authority";
import {panelVariants} from "./paper";

export type PaperVariants = VariantProps<typeof panelVariants>;

export const panelRoundVariants = ['lg', 'no', 'xs'] as const satisfies Omit<PaperVariants['round'], 'null' | 'undefined'>;

export const panelOutlineVariants = [
  '700',
  '500',
  '200',
  '50',
] as const satisfies Omit<PaperVariants['outline'], 'null' | 'undefined'>;
