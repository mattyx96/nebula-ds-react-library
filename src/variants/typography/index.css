.display1 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-display1);
}

.display2 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-display2);
}

.header1 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header1);
}

.header2 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header2);
}

.header3 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header3);
}

.header4 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header4);
}

.header5 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header5);
}

.header6 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-header6);
}

.body1 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-body1);
}

.body2 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-body2);
}

.body3 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-body3);
}

.body4 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-body4);
}

.body5 {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-body5);
}

.button {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-button);
}

.caption {
  color: var(--nb-background-contrast-primary-500);
  font: var(--nb-caption);
}
