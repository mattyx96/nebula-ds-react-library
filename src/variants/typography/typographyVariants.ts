import {cva} from "class-variance-authority";

export const typographyVariants = cva(
  '',
  {
    variants: {
      variant: {
        display1: 'display1',
        display2: 'display2',
        header1: 'header1',
        header2: 'header2',
        header3: 'header3',
        header4: 'header4',
        header5: 'header5',
        header6: 'header6',
        body1: 'body1',
        body2: 'body2',
        body3: 'body3',
        body4: 'body4',
        body5: 'body5',
        button: 'button',
        caption: 'caption',
      },
      weight: {
        default: '',
        light: 'font-light',
        bold: 'font-bold',
        bolder: 'font-extrabold',
      },
    },
    defaultVariants: {
      weight: 'default',
    },
  }
);
