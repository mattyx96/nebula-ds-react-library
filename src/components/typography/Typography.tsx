import {type ComponentPropsWithRef, createElement, ReactNode} from 'react';
import {clsxMerge} from '../../common/utils/classNameUtils';
import {TypographyVariants} from "../../variants/typography";
import {typographyVariants} from "../../variants/typography/typographyVariants";

import '../../variants/typography/index.css'

export type Tags = 'p' | 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'label';

type TypographyElementProps<T extends Tags> = ComponentPropsWithRef<T>;

type TypographyPropsBase<T extends Tags> = TypographyElementProps<T> & TypographyVariants;

export type TypographyWithTextProps<T extends Tags> = TypographyPropsBase<T> & {
  text?: string;
  component?: T;
  children?: ReactNode;
  className?: string;
};

export type TypographyProps<T extends Tags> = TypographyWithTextProps<T>;

export const Text = <T extends Tags = 'p'>(props: TypographyProps<T>) => {
  const {text, component, className, children, ...rest} = props;

  return createElement(component || 'p', {
    ...{
      className: clsxMerge(
        typographyVariants({
          variant: props.variant,
        }),
        className
      )
    }, ...rest
  }, children || text);
}

