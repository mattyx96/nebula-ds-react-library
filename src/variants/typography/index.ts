import {VariantProps} from "class-variance-authority";
import {typographyVariants} from "./typographyVariants.ts";
import {Tags} from "../../components/typography/Typography.tsx";

export type TypographyVariants = VariantProps<typeof typographyVariants>;

export const typographyVariantVariants = [
  'display1',
  'display2',
  'header1',
  'header2',
  'header3',
  'header4',
  'header5',
  'header6',
  'body1',
  'body2',
  'body3',
  'body4',
  'body5',
  'button',
  'caption'
] as const satisfies Omit<TypographyVariants['variant'], 'null' | 'undefined'>

export const typographyWeightVariants = [
  'default',
  'light',
  'bold',
  'bolder',
] as const satisfies Omit<TypographyVariants['weight'], 'null' | 'undefined'>

export const typographyComponentVariants = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'p',
  'span',
] satisfies Tags[]
