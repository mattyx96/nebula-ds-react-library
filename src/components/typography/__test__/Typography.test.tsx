import {render} from '@testing-library/react';
import {Tags, Text} from '../Typography.tsx';
import {describe, expect, it} from 'vitest';
import {typographyVariantVariants, typographyWeightVariants} from "../../../variants/typography";

describe('Text', () => {
  it('renders correctly', () => {
    const {container} = render(<Text text="Sample Text"/>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with different variants', () => {
    typographyVariantVariants.forEach((variant) => {
      const {container} = render(<Text text={`Text with ${variant}`} variant={variant}/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders correctly with different components', () => {
    const components: Tags[] = ['p', 'span', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
    components.forEach((component) => {
      const {container} = render(<Text text={`Text as ${component}`} component={component}/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders correctly with different weights', () => {
    typographyWeightVariants.forEach((weight) => {
      const {container} = render(<Text text={`Text with weight ${weight}`} weight={weight}/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders correctly with children', () => {
    const {container} = render(<Text>Child Text</Text>);
    expect(container.firstChild).toHaveTextContent('Child Text');
  });

  it('handles className prop correctly', () => {
    const {container} = render(<Text text="Class Name Text" className="custom-class"/>);
    expect(container.firstChild).toHaveClass('custom-class');
  });
});
