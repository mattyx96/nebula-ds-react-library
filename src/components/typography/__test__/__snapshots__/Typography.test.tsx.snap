// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > renders correctly 1`] = `
<p
  class=""
>
  Sample Text
</p>
`;

exports[`Text > renders correctly with different components 1`] = `
<p
  class=""
>
  Text as p
</p>
`;

exports[`Text > renders correctly with different components 2`] = `
<span
  class=""
>
  Text as span
</span>
`;

exports[`Text > renders correctly with different components 3`] = `
<h1
  class=""
>
  Text as h1
</h1>
`;

exports[`Text > renders correctly with different components 4`] = `
<h2
  class=""
>
  Text as h2
</h2>
`;

exports[`Text > renders correctly with different components 5`] = `
<h3
  class=""
>
  Text as h3
</h3>
`;

exports[`Text > renders correctly with different components 6`] = `
<h4
  class=""
>
  Text as h4
</h4>
`;

exports[`Text > renders correctly with different components 7`] = `
<h5
  class=""
>
  Text as h5
</h5>
`;

exports[`Text > renders correctly with different components 8`] = `
<h6
  class=""
>
  Text as h6
</h6>
`;

exports[`Text > renders correctly with different variants 1`] = `
<p
  class="display1"
  variant="display1"
>
  Text with display1
</p>
`;

exports[`Text > renders correctly with different variants 2`] = `
<p
  class="display2"
  variant="display2"
>
  Text with display2
</p>
`;

exports[`Text > renders correctly with different variants 3`] = `
<p
  class="header1"
  variant="header1"
>
  Text with header1
</p>
`;

exports[`Text > renders correctly with different variants 4`] = `
<p
  class="header2"
  variant="header2"
>
  Text with header2
</p>
`;

exports[`Text > renders correctly with different variants 5`] = `
<p
  class="header3"
  variant="header3"
>
  Text with header3
</p>
`;

exports[`Text > renders correctly with different variants 6`] = `
<p
  class="header4"
  variant="header4"
>
  Text with header4
</p>
`;

exports[`Text > renders correctly with different variants 7`] = `
<p
  class="header5"
  variant="header5"
>
  Text with header5
</p>
`;

exports[`Text > renders correctly with different variants 8`] = `
<p
  class="header6"
  variant="header6"
>
  Text with header6
</p>
`;

exports[`Text > renders correctly with different variants 9`] = `
<p
  class="body1"
  variant="body1"
>
  Text with body1
</p>
`;

exports[`Text > renders correctly with different variants 10`] = `
<p
  class="body2"
  variant="body2"
>
  Text with body2
</p>
`;

exports[`Text > renders correctly with different variants 11`] = `
<p
  class="body3"
  variant="body3"
>
  Text with body3
</p>
`;

exports[`Text > renders correctly with different variants 12`] = `
<p
  class="body4"
  variant="body4"
>
  Text with body4
</p>
`;

exports[`Text > renders correctly with different variants 13`] = `
<p
  class="body5"
  variant="body5"
>
  Text with body5
</p>
`;

exports[`Text > renders correctly with different variants 14`] = `
<p
  class="button"
  variant="button"
>
  Text with button
</p>
`;

exports[`Text > renders correctly with different variants 15`] = `
<p
  class="caption"
  variant="caption"
>
  Text with caption
</p>
`;
